drop-transfer/src/check.rs

```diff
@@ -7,51 +7,91 @@ use tokio_util::sync::CancellationToken;
 use crate::{protocol, service::State, tasks::AliveGuard, IncomingTransfer, Transfer};
 
 pub(crate) fn spawn(
+    refresh_trigger: tokio::sync::watch::Receiver<()>,
     state: Arc<State>,
     xfer: Arc<IncomingTransfer>,
     logger: Logger,
     guard: AliveGuard,
     stop: CancellationToken,
-) -> tokio::task::JoinHandle<()> {
+) {
     let id = xfer.id();
-    let job = run(state, xfer, logger.clone());
 
     tokio::spawn(async move {
         let _guard = guard;
+        let mut refresh_trigger = refresh_trigger;
+
+        let logger_clone = logger.clone();
+        let task = async move {
+            let logger = logger_clone;
+            loop {
+                let cf = run(&state, xfer.clone(), &logger).await;
+
+                if cf.is_break() {
+                    break;
+                }
+
+                #[allow(clippy::let_underscore_must_use)]
```

Comment:
I think we've disabled this lint after our discussion under the rust 1.72.1 bump PR